src/lib/users/leaf_blk_device.rb

```diff
@@ -0,0 +1,121 @@
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "json"
+require "yast2/execute"
+
+module Y2Users
+  # This class represents a block device reported by `lsblk` as a leaf node
+  #
+  # As we cannot use a devicegraph for the time being, this class extracts the
+  # block devices information from the lsblk command. Only the leaf devices
+  # are taken into account and it only includes information which is relevant
+  # for the {Y2Users::Widgets::DiskSelector} widget.
+  class LeafBlkDevice
```

Comment:
Would this class also make sense outside of `yast2-users`? Maybe we could share the class...

Reply:
That's a fair point but I would say that we should use libstorage-ng instead. But the problem is that we cannot do the probing again at this point (it will modify the probing/staging devicegraphs). I will open the discussion about introducing some libstorage-ng changes in order to be able to do that, but it will take time.